storage/replica_raftstorage.go

```diff
@@ -30,16 +30,24 @@ import (
 	"github.com/gogo/protobuf/proto"
 )
 
+// Most calls to raft.RawNode require that the replica read lock is held.
```

Comment:
Also the read/write lock distinction doesn't matter here. `RawNode` is not thread safe, so all access to it must be guarded by an exclusive lock. The reason I wanted a comment here was to emphasize that the lock that guards `raftGroup` must be the same as the lock that guards all the inner fields; we can't use two separate locks.